File: Strings/3713_Longest_Balanced_Substring_I/Solution.java
class Solution {
    public int longestBalanced(String s) {
        int n = s.length();
        int maxLength = 0;

        for (int start = 0; start < n; start++) {
            int[] frequency = new int[26];
            int distinctCount = 0;
            int maxFrequency = 0;

            for (int end = start; end < n; end++) {
                int index = s.charAt(end) - 'a';

                if (frequency[index] == 0) {
                    distinctCount++;
                }

                frequency[index]++;
                maxFrequency = Math.max(maxFrequency, frequency[index]);

                int currentLength = end - start + 1;

                if (maxFrequency * distinctCount == currentLength) {
                    maxLength = Math.max(maxLength, currentLength);
                }
            }
        }

        return maxLength;
    }
}

File: Strings/3713_Longest_Balanced_Substring_I/solution.py
class Solution:
    def longestBalanced(self, s: str) -> int:
        n = len(s)
        max_length = 0

        for start in range(n):
            freq = [0] * 26
            distinct = 0
            max_freq = 0

            for end in range(start, n):
                idx = ord(s[end]) - ord('a')

                if freq[idx] == 0:
                    distinct += 1

                freq[idx] += 1
                max_freq = max(max_freq, freq[idx])

                length = end - start + 1

                if max_freq * distinct == length:
                    max_length = max(max_length, length)

        return max_length
